agent/consul/server_serf.go

```diff
@@ -386,6 +386,11 @@ func (s *Server) maybeBootstrap() {
 		return
 	}
 
+	if s.config.ReadReplica {
```

Comment:
I put this after the check above so that it'll converge on `s.config.BootstrapExpect = 0` at the same time as all of the other nodes so the rest of the loop below can handle the `BootstrapExpect` equality check logic properly.